fontforge/python.c

```diff
@@ -1717,50 +1608,42 @@ static PyObject *PyFF_ask(PyObject *UNUSED(self), PyObject *args) {
 return( NULL );
     }
 
-    if ( !PyArg_ParseTuple(args,"esesO|ii","UTF-8", &title, "UTF-8", &quest, &answero, &def, &cancel) )
+    if ( !PyArg_ParseTuple(args,"ssO|ii", &title, &quest, &answero, &def, &cancel) )
 return( NULL );
-    if ( !PySequence_Check(answero) || STRING_CHECK(answero)) {
+    if ( !PySequence_Check(answero) || PyUnicode_Check(answero)) {
 	PyErr_Format(PyExc_TypeError, "Expected a tuple of strings for the third argument");
-	PyMem_Free(title);
-	PyMem_Free(quest);
 return( NULL );
     }
     cnt = PySequence_Size(answero);
-    answers = calloc(cnt+1, sizeof(char *));
+    answers = calloc(cnt+1, sizeof(const char *));
     if ( cancel==-1 )
 	cancel = cnt-1;
     if ( cancel<0 || cancel>=cnt || def<0 || def>=cnt ) {
 	PyErr_Format(PyExc_ValueError, "Value out of bounds for 4th or 5th argument");
-	PyMem_Free(title);
-	PyMem_Free(quest);
 	free(answers);
 return( NULL );
     }
     for ( i=0; i<cnt; ++i ) {
-        PyObject *utf8_name = PYBYTES_UTF8(PySequence_GetItem(answero,i));
-	if ( utf8_name==NULL ) {
-	    PyMem_Free(title);
-	    PyMem_Free(quest);
-	    FreeStringArray( i, answers );
-return( NULL );
-	}
-	answers[i] = copy(PyBytes_AsString(utf8_name));
-	Py_DECREF(utf8_name);
+        PyObject *item = PySequence_GetItem(answero, i);
+        answers[i] = PyUnicode_AsUTF8(item);
+        Py_XDECREF(item);
```

Comment:
I haven't checked in depth whether or not it's ok (decrementing the reference while holding the string rep), but my reading of the docs for PyUnicode_AsUTF8 (combined with assuming the sequence is a list/tuple) suggests that it should be, as long as that reference to the object in the list/tuple still exists